File: app/src/main/java/com/botijasoftware/utils/IndexBuffer.java
package com.botijasoftware.utils;

import java.nio.Buffer;

public abstract class IndexBuffer {

	protected int maxelements;
	protected int size;
	protected int type;
	protected int format;
	protected int elementsize;
	
	public IndexBuffer(int type, int maxelements) {
		this.type = type;
		this.maxelements = maxelements;
		this.size = maxelements;
	}
		
	public int getSize() {
		return size;
	}
	
	public void setSize(int size) {
		if (size >= 0) {
			this.size = size;
		}
		if (this.size > maxelements)
			this.size = maxelements;
	}
	
	public int getMaxElements() {
		return maxelements;
	}
	
	public int getType() {
		return type;
	}
	
	public int getFormat() {
		return format;
	}
	
	public abstract Buffer getBuffer();
	
	public void put(int index, byte value) {}
	public void put(byte[] data) {}
	public void put(int index, short value) {}
	public void put(short[] data) {}
	public void put(int index, int value) {}
	public void put(int[] data) {}
	public void put(byte value) {}
	public void put(short value) {}
	public void put(int value) {}
	
	public void position(int i) {}
	public int position() {return 0;}


}

File: app/src/main/java/com/botijasoftware/utils/IndexBuffer.kt
package com.botijasoftware.utils

import java.nio.Buffer

abstract class IndexBuffer(var type: Int, maxelements: Int) {

    var maxElements: Int = 0
    var size: Int = 0
        set(value) {
            if (value >= 0) {
                field = value
            }
            if (field > maxElements)
                field = maxElements
        }

    var format: Int = 0
    protected var elementsize: Int = 0

    init {
        this.maxElements = maxelements
        this.size = maxelements
    }


    abstract val buffer: Buffer

    open fun put(index: Int, value: Byte) {}
    open fun put(data: ByteArray) {}
    open fun put(index: Int, value: Short) {}
    open fun put(data: ShortArray) {}
    open fun put(index: Int, value: Int) {}
    open fun put(data: IntArray) {}
    open fun put(value: Byte) {}
    open fun put(value: Short) {}
    open fun put(value: Int) {}

    open fun position(i: Int) {}
    open fun position(): Int {
        return 0
    }


}